### DISCUSSION TITLE
Examples not reproducible because of exports

### DISCUSSION THREAD

**Discussion by sebix** [textX/textX#331] [Q&A]
Hi,

currently the documentation/examples of textX is not [reproducible](https://reproducible-builds.org/) because the export of the robot example contains some kind of random(?) identifier:

Here's a diff between two builds:
```diff
/usr/share/doc/packages/python-textX-doc/examples/robot/robot_meta.dot differs (ASCII text)
--- old//usr/share/doc/packages/python-textX-doc/examples/robot/robot_meta.dot	2018-07-06 00:00:00.000000000 +0000
+++ new//usr/share/doc/packages/python-textX-doc/examples/robot/robot_meta.dot	2018-07-06 00:00:00.000000000 +0000
@@ -28,7 +28,7 @@
 
 93824998766232[ label="{ID|}"]
 
-93824998689320[ label="{STRING|}"]
+93824999182760[ label="{STRING|}"]
 
 93824999619672[ label="{BOOL|}"]
 
@@ -44,7 +44,7 @@
 93824999420088 -> 93824999631000 [dir=back]
 93824999420088 -> 93824999619672 [dir=back]
 93824999420088 -> 93824998766232 [dir=back]
-93824999420088 -> 93824998689320 [dir=back]
+93824999420088 -> 93824999182760 [dir=back]
 
 93824999246744[ label="{*OBJECT|}"]
 93824999246744 -> 93824999420088 [dir=back]
```
Would it be possible to make the export deterministic?

**Comment by igordejanovic** [comment]
Hi @sebix. For identifiers in exported dot files `id` function is used which returns a unique identifier of Python object and is implementation specific (usually related to the address in memory), so yes it changes from run to run. My question is what are you trying to achieve? Examples are not part of the build, at least they shouldn't be.
In order to ids in dot files to be deterministic we would need something model/meta-model related that could be uniquely calculated for each model object and would change only when model/meta-model definition change.

**Comment by bmwiedemann** [comment]
[In our case](https://build.opensuse.org/package/view_file/openSUSE:Factory/python-textX/python-textX.spec), `examples/robot/robot_meta.dot` is created by `py.test-3.6` in the `%check` section
and installed along with other examples via the `%doc` macro.

https://en.opensuse.org/openSUSE:Reproducible_Builds has useful links on the topic

The final goal is to have every package build able to produce bit-identical results anywhere anytime.
Dropping unnecessary files is one valid option to get there.
Replacing `id` with some stable digest/hash function could be a nice solution, too.

**Comment by igordejanovic** [comment]
I see. `dot` files in examples are generated during test as all examples are run to verify that they don't break. The produced `dot` files are not considered a part of the build, they are byproduct. Do you have an option in `%doc` macro to exclude `.dot` files from examples folder?

**Comment by bmwiedemann** [comment]
proposed https://build.opensuse.org/request/show/651138 to drop the .dot example files
